### ISSUE TITLE
LLVM.jl 7.2 seems to break Enzyme

### ISSUE THREAD

**Issue by ptiede** [EnzymeAD/Enzyme.jl#1496]
I just saw this appear on my CI. The recent `LLVM.jl@7.2` release breaks Enzyme.jl.

When I add Enzyme in a clean environment, I get

```
ERROR: LoadError: UndefVarError: `InstructionMetadataDict` not defined
Stacktrace:
 [1] getproperty(x::Module, f::Symbol)
   @ Base ./Base.jl:31
 [2] top-level scope
   @ ~/.julia/packages/Enzyme/F71IJ/src/api.jl:107
 [3] include(mod::Module, _path::String)
   @ Base ./Base.jl:495
 [4] include(x::String)
   @ Enzyme ~/.julia/packages/Enzyme/F71IJ/src/Enzyme.jl:1
 [5] top-level scope
   @ ~/.julia/packages/Enzyme/F71IJ/src/Enzyme.jl:33
 [6] include
   @ ./Base.jl:495 [inlined]
 [7] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
   @ Base ./loading.jl:2222
 [8] top-level scope
   @ stdin:3
in expression starting at /home/ptiede/.julia/packages/Enzyme/F71IJ/src/api.jl:1
in expression starting at /home/ptiede/.julia/packages/Enzyme/F71IJ/src/Enzyme.jl:1
in expression starting at stdin:
```

everything is fine if I use `LLVM.jl@v7.1`. I am not sure if this is the right repo, but I thought I would report it here.

**Comment by ptiede** [comment]
Looking into this a bit, it appears that `InstructionMetadataDict` was removed in 7.2, even though the name was exported. Maybe this is an issue with LLVM.jl then?

**Comment by wsmoses** [comment]
https://github.com/maleadt/LLVM.jl/pull/414/files apparently

**Comment by ptiede** [comment]
https://github.com/maleadt/LLVM.jl/issues/415

**Comment by vchuravy** [comment]
Pushed a fix to the registery, and https://github.com/JuliaRegistries/General/pull/108404 will fix the code itself.
@maleadt also pushed a deprecation.

```
(jl_o19IhO) pkg> add Enzyme
   Resolving package versions...
    Updating `/tmp/jl_o19IhO/Project.toml`
  [7da242da] + Enzyme v0.12.9
    Updating `/tmp/jl_o19IhO/Manifest.toml`
  [fa961155] + CEnum v0.5.0
  [7da242da] + Enzyme v0.12.9
  [f151be2c] + EnzymeCore v0.7.3
  [e2ba6199] + ExprTools v0.1.10
  [61eb1bfa] + GPUCompiler v0.26.5
  [692b3bcd] + JLLWrappers v1.5.0
⌅ [929cbde3] + LLVM v7.1.0
```

**Comment by wsmoses** [comment]
Oh wait you released enzyme 0.12.10 there were some experimental things
there that I hadn’t sufficiently validated on downstream (specifically
changing to blas tblgen for forward mode).

**Comment by vchuravy** [comment]
... welp, we will get real-world testing.

Since the PR was called "Test forward blas" and had bumped the version I assumed it was fine.

We can release `0.12.11` with https://github.com/EnzymeAD/Enzyme.jl/pull/1490/files#diff-365fc3b77837a3d3be094efbadc976064a26a40534c0adfb9ca65dc98819134eR4635 being reverted?
